### PR TITLE
Use tini as the entry point to handle signals correctly

### PR DESCRIPTION
**Pull request by Moredread**
Proposal to solve issue #8

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,10 +12,17 @@ RUN apt-get update && \
...
+ && gpg --verify /tini.asc
+RUN chmod +x /tini
+
 COPY entrypoint.sh /sbin/entrypoint.sh
 RUN chmod 755 /sbin/entrypoint.sh
 
 EXPOSE 3142/tcp
 VOLUME ["${APT_CACHER_NG_CACHE_DIR}"]
-ENTRYPOINT ["/sbin/entrypoint.sh"]
+ENTRYPOINT ["/tini", "--", "/sbin/entrypoint.sh"]
```
Comment (sameersbn) [inline Dockerfile]: can you invoke `tini` from the `entrypoint.sh` (https://github.com/sameersbn/docker-apt-cacher-ng/blob/master/entrypoint.sh#L37) script instead?

File: Dockerfile [hunk 2]
```diff
@@ -12,10 +12,17 @@ RUN apt-get update && \
  && sed 's/# PassThroughPattern:.*this would allow.*/PassThroughPattern: .* #/' -i /etc/apt-cacher-ng/acng.conf \
  && rm -rf /var/lib/apt/lists/*
 
+ENV TINI_VERSION v0.14.0
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini.asc /tini.asc
+RUN gpg --keyserver ha.pool.sks-keyservers.net --recv-keys 595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7 \
+ && gpg --verify /tini.asc
+RUN chmod +x /tini
```
Comment (sameersbn) [inline Dockerfile]: I would prefer `tini` being installed at `/usr/local/bin/`

File: Dockerfile [hunk 3]
```diff
@@ -12,10 +12,17 @@ RUN apt-get update && \
  && sed 's/# PassThroughPattern:.*this would allow.*/PassThroughPattern: .* #/' -i /etc/apt-cacher-ng/acng.conf \
  && rm -rf /var/lib/apt/lists/*
 
+ENV TINI_VERSION v0.14.0
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
```
Comment (sameersbn) [inline Dockerfile]: I would prefer the `tini` being installed in a single Dockerfile instruction, eg. https://github.com/bitnami/minideb-extras/blob/master/jessie/Dockerfile#L25-L32

**Comment by Moredread** [comment]
Sure, I'll make the changes ASAP

**Comment by Moredread** [comment]
I've done the changes, hope everything is fine now. :)

**Comment by Moredread** [comment]
@sameersbn had you time already to look at the changes?

**Comment by sameersbn** [comment]
sorry for the delay. the changes have been merged :+1:

**Comment by sameersbn** [comment]
please note that i've removed `tini` from the image. You should use the `--init` flag so that docker daemon injects a init process into the container that correctly reaps process and handles signals.

**Pull request merged**